src/driver/amdxdna/aie2_msg_priv.h

```diff
@@ -352,6 +352,18 @@ struct cmd_chain_slot_execbuf_cf {
 	u32 args[] __counted_by(arg_cnt);
 };
 
+#define slot_dpu_has_space(offset, payload_size) \
+	(MAX_CHAIN_CMDBUF_SIZE - ((offset) + (payload_size)) > \
+	 offsetof(struct cmd_chain_slot_dpu, args[0]))
```

Comment:
struct cmd_chain_slot_dpu is of variable size, do you want to use 40 DWORD for the calculation?

Reply:
I want to use the accurate size here to optimize the DMA. This macro just make sure the following memcpy() doesn't write out of bounds.

Your comment reminds me, when calling into this macro, the payload_len should minus sizeof(*sd).